CRM/Event/BAO/Event.php

```diff
@@ -1099,6 +1099,7 @@ public static function sendMail($contactID, $values, $participantId, $isTest = F
 
     if ($values['event']['is_email_confirm'] || $returnMessageText) {
       list($displayName, $email) = CRM_Contact_BAO_Contact_Location::getEmailDetails($contactID);
+      $email = CRM_Utils_Array::valueByRegexKey('/^email-/', $participantParams) ?? $email;
```

Comment:
I'm assuming this is intentional that it only falls back to primary if it's truly missing as opposed to the empty string. I think I agree with that because if the field on the profile is optional, and they leave it blank, then it might be unexpected that they receive an email at their primary. If that was your intent, it might be useful to have a comment explaining it's on purpose. But I note this is different than the `empty()` check you have down in ThankYou.php, which would fall back to primary if left blank on the form (I think - haven't tested that part yet, just this first part.)

Reply:
@jitendrapurohit I submitted a patch today for the same issue, I'll incorporate @demeritcowboy's suggestion.